internal/js/modules/k6/browser/common/frame_manager.go

```diff
@@ -599,14 +599,21 @@ func (m *FrameManager) NavigateFrame(frame *Frame, url string, parsedOpts *Frame
 	timeoutCtx, timeoutCancelFn := context.WithTimeout(m.ctx, parsedOpts.Timeout)
 	defer timeoutCancelFn()
 
-	newDocIDCh := make(chan string, 1)
+	var newDocumentID string
+	newDocIDIsReadyCh := make(chan struct{})
```

Comment:
The problem here was that there might be more than one `NavigationEvent`. For instance, in the case where you try to navigate to a blocked IP or hostname.

And, if there's more than one event, the line `newDocID := <-newDocIDCh` below is blocking for all the events except the first one. But the first one isn't necessarily the one we're really interested on (the one for which we return `true`).

So, instead, I only use the channel as a synchronization primitive (see I `close` it below, once we have the document identifier), while the value is stored in a standard variable, read by the closure, instead of using a channel for sending such value.